### PR TITLE
Feature/ add substrate queries

### PR DESCRIPTION
**Pull request by TxCorpi0x**
## Description

Query methods are available in the `query.go`  in the `substrate` package and it's methods are being used by the `substrate` package and the rest of the packages of the relayer code-base.

## Added Methods to `query.go`

- [x] `QueryBalance`
- [x] `QueryBalanceWithAddress`
- [x] `QueryClientStateResponse`
- [x] `QueryClientState`
- [x] `QueryClientConsensusState`
- [x] `QueryUpgradeProof`
- [x] `QueryUpgradedClient`
- [x] `QueryUpgradedConsState`
- [x] `QueryConsensusState`
- [x] `QueryConnection`
- [x] `QueryConnections`
- [x] `QueryConnectionsUsingClient`
- [x] `GenerateConnHandshakeProof`
- [x] `QueryChannel`
- [x] `QueryChannelClient`
- [x] `QueryConnectionChannels`
- [x] `QueryChannels`
- [x] `QueryPacketCommitments`
- [x] `QueryPacketAcknowledgements`
- [x] `QueryUnreceivedPackets`
- [x] `QuerySendPacket`
- [x] `QueryRecvPacket`
- [x] `QueryUnreceivedAcknowledgements`
- [x] `QueryNextSeqRecv`
- [x] `QueryPacketCommitment`
- [x] `QueryPacketAcknowledgement`
- [x] `QueryPacketReceipt`
- [x] `QueryLatestHeight`
- [x] `QueryHeaderAtHeight`
- [x] `QueryDenomTrace`
- [x] `QueryDenomTraces`
- [x] `QueryConsensusStateABCI`

### CODE REVIEW

**Comment by blasrodri** [comment]
@jtieri 
I'm getting after solving conflicts
```
go: finding module for package github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx
github.com/cosmos/relayer/v2/relayer/chains/cosmos imports
	github.com/tendermint/tendermint/rpc/core/types: module github.com/tendermint/tendermint@latest found (v0.35.9), but does not contain package github.com/tendermint/tendermint/rpc/core/types
github.com/cosmos/relayer/v2/relayer/chains/substrate imports
	github.com/cosmos/ibc-go/v5/modules/light-clients/11-beefy/types: module github.com/cosmos/ibc-go/v5@latest found (v5.0.0-rc1), but does not contain package github.com/cosmos/ibc-go/v5/modules/light-clients/11-beefy/types
github.com/cosmos/relayer/v2/relayer/chains/cosmos imports
(truncated)
```
any idea on how to fix it?

File: relayer/chains/substrate/query.go
Comment (jtieri) [inline relayer/chains/substrate/query.go:66]: After some light Googling I see that Polkadot has an unbonding period of 28 days, my naive understanding is that all stake in Polkadot is locked up on the relay chain so does that mean the 28 day unbonding period applies in all cases no matter what Parachain may be actually connecting to IBC?

If so could this be hardcoded to 28 days?

File: relayer/chains/substrate/query.go
Comment (jtieri) [inline relayer/chains/substrate/query.go:55]: Looks like this TODO may no longer be needed

File: relayer/chains/substrate/query.go
Comment (jtieri) [inline relayer/chains/substrate/query.go:42]: This would change the SubstrateProvider's configured key in memory but not on disk so later operations could be referencing two different keys, the one on disk in the config file or the one now referenced in memory. As a consumer of this API it would be an unexpected side effect for a method named `QueryBalance` to modify state instead of just querying some data. In the cosmos implementation we just throw an error if the `keyName` argument is invalid for any reason whether it's an empty string or there is no key that matches the name.

**Review by jtieri — APPROVED** [review]
Just had one question and a couple nit picks but this looks good otherwise. I'll approve and leave it up to you if you wanna address the few comments I left now or later.

File: relayer/chains/substrate/query.go
Comment (oshorefueled) [reply relayer/chains/substrate/query.go:42]: Yeah, that makes sense. I'll resolve it in another PR.

File: relayer/chains/substrate/query.go
Comment (oshorefueled) [reply relayer/chains/substrate/query.go:66]: yes, the unbonding period would apply to all parachains secured by the relaychain. However, the duration of the unbonding period is dependent on the network. For instance Kusama has a 7 day unbonding period. So we'll probably need to use a switch statement for that. I'll add it in another  PR.

**Pull request merged**
